backend/wmg/data/extract.py

```diff
@@ -1,11 +1,56 @@
 # TODO generalize
+import os
+
+from backend.corpora.common.corpora_orm import DatasetArtifactFileType
+from backend.corpora.common.entities import Dataset, DatasetAsset
+from backend.corpora.common.utils.db_session import db_session_manager
+
+# TODO make dict of ontology ids and human readable names, store as a const elsewhere (in rds for easy updates?)
+included_assay_ontology_ids = [
+    "EFO:0008722",
+    "EFO:0010010",
+    "EFO:0010550",
+    "EFO:0010961",
+    "EFO:0030002",
+    "EFO:0009901",
+    "EFO:0011025",
+    "EFO:0009899",
+    "EFO:0009900",
+    "EFO:0009922",
+    "EFO_0030003",
+    "EFO:0030004",
+    "EFO:0008919",
+]
+
+
 def get_s3_uris():
     """
     Retrieve list of s3 uris for datasets included in the wmg cube
     """
-    pass
+    with db_session_manager() as session:
+
+        dataset_ids = []
+        published_dataset_non_null_assays = (
+            session.query(Dataset.table.id, Dataset.table.assay)
+            .filter(
+                Dataset.table.assay != "null",
+                Dataset.table.published == "TRUE",
+                Dataset.table.is_primary_data == "PRIMARY",
+                Dataset.table.collection_visibility == "PUBLIC",
+                Dataset.table.tombstone == "f",
+            )
+            .all()
+        )
+        for dataset in published_dataset_non_null_assays:
+            if dataset[1]["ontology_term_id"] in included_assay_ontology_ids:
+                dataset_ids.append(dataset[0])
```

Comment:
I'm also wondering if the ORM supports the SQL `IN` operator? Potentially could have the db filter down to the assay types we want to retrieve.

Reply:
tbh I hate list comprehensions, I find them to be more difficult to grok at a glance then writing out the full loop

Reply:
I'm not 100% sure but I dont think we can use the IN operator. The assay data is a json blob that contains a list of dicts so I need to pull them out to read them/do a comparison. 
When doing it directly in SQL this works
```select id, items
from dataset,
     jsonb_to_recordset(dataset.assay) as items(ontology_term_id text)
where assay != 'null'
  and published = 'TRUE'
  and is_primary_data = 'PRIMARY'
  and collection_visibility = 'PUBLIC'
  and tombstone = 'f'
  and items.ontology_term_id in ('EFO:0008722','EFO:0010010', 'EFO:0010550', 'EFO:0010961', 'EFO:0030002', 'EFO:0009901', 'EFO:0011025', 'EFO:0009899', 'EFO:0009900', 'EFO:0009922', 'EFO:0030003', 'EFO:0030004', 'EFO:0008919');```
  but I couldnt find a good way to do it using the orm so I thought it made more sense to just split it into two steps for simplicity/to avoid rabbit holing when there was an easy workaround